res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1383 @@
+            group: "[Library]",
+            inKey: "MoveVertical",
+            inValueScale: DDJSR2.getRotaryDelta
+        }),
+        press: new components.Button({
+            group: "[Library]",
+            key: "GoToItem"
+        }),
+        shiftPress: new components.Button({
+            group: "[Library]",
+            input: function(channel, control, value, status, group) {
+                if (this.isPress(channel, control, value, status, group)) {
+                    Browser.browseKnob.trackColorCycleEnabled = true;
+                    return;
+                }
+                // On release, not press. Allows for track color cycling.
+                Browser.browseKnob.trackColorCycleEnabled = false;
+                if (Browser.browseKnob.trackColorCycleHappened) {
+                    Browser.browseKnob.trackColorCycleHappened = false;
+                    return;
+                }
+                if (!engine.getValue("[PreviewDeck1]", "play")) {
+                    engine.setValue("[PreviewDeck1]", "LoadSelectedTrackAndPlay", 1);
+                } else {
+                    script.triggerControl("[PreviewDeck1]", "stop");
+                }
+            }
+        }),
+        shiftTurn: new components.Encoder({
+            group: "[PreviewDeck1]",
+            input: function(channel, control, value, _status, _group) {
+                const rotateValue = DDJSR2.getRotaryDelta(value);
+                if (!Browser.browseKnob.trackColorCycleEnabled) {
+                    engine.setValue("[PreviewDeck1]", "beatjump", rotateValue*8);
+                } else {
+                    const key = (rotateValue > 0) ? "track_color_next" : "track_color_prev";
+                    engine.setValue("[Library]", key, 1.0);
+                    Browser.browseKnob.trackColorCycleHappened = true;
+                }
+            }
+        }),
+    });
+    this.loadButtons = [];
+    for (let channelOffset = 0; channelOffset < 4; channelOffset++) {
+        const group = `[Channel${channelOffset+1}]`;
+        this.loadButtons[channelOffset] = new components.Button({
+            group: group,
+            key: "LoadSelectedTrack",
+            midi: [0x9B, 0x0+channelOffset]
+        });
+        this.ejectTrackButton = new components.Button({
+            inKey: "eject",
+            outKey: "track_loaded",
+            midi: [0x96, 0x46+channelOffset]
+        });
+    }
+
+    this.sortLibrary = function(channel, control, value, _status, _group) {
+        if (value === 0) {
+            return;
+        }
+
+        let sortColumn;
+        switch (control) {
+        case 0x58:  // BPM
+            sortColumn = script.LIBRARY_COLUMNS.BPM;
+            break;
+        case 0x59:  // ARTIST
+            sortColumn = script.LIBRARY_COLUMNS.ARTIST;
+            break;
+        default:
+            // unknown sort column
+            return;
+        }
+        engine.setValue("[Library]", "sort_column", sortColumn);
+    };
```

Comment:
Obviously the duplicated initial definitions aren't great either and could be trivially solved by simply subclassing, but thats rather tedious unfortunately, so I'm okay with the repetition here.
```suggestion
    this.sortLibraryBpm = new components.Button({
        inKey: "sort_column",
        group: "[Library]",
        type: components.Button.prototype.types.toggle;
        inToggle: function() {
            this.inSetValue(script.LIBRARY_COLUMNS.BPM);
        }
    });
    this.sortLibraryArtist= new components.Button({
        inKey: "sort_column",
        group: "[Library]",
        type: components.Button.prototype.types.toggle;
        inToggle: function() {
            this.inSetValue(script.LIBRARY_COLUMNS.ARTIST);
        }
    });
```